test/test-moc.js

```diff
@@ -143,6 +143,18 @@ assert.deepStrictEqual(Motoko.check("bad.mo"), {
       message:
         "unexpected end of input, expected one of token or <phrase> sequence:\n  <exp_bin(ob)>",
     },
+    {
+        source: 'bad.mo',
+        severity: 1,
+        range: {
+          start: { line: 0, character: 0 },
+          end: { line: 0, character: 0 }
+        },
+        code: 'M0057',
+        category: 'type',
+        message: 'unbound variable __error_recovery_var__'
```

Comment:
Maybe we should plug holes with values of bottom type None instead? 
Maybe `loop{}` or a special prim of type None?